jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java

```diff
@@ -531,16 +535,43 @@ public void setMinResponseDataRate(long bytesPerSecond)
         _minResponseDataRate = bytesPerSecond;
     }
 
+    /**
+     * @see #getSetCookieCompliance()
+     * @return The CookieCompliance used for parsing <code>Cookie</code> headers.
+     */
     public CookieCompliance getCookieCompliance()
     {
         return _cookieCompliance;
     }
-    
+
+    /**
+     * @see #getCookieCompliance()
+     * @return The CookieCompliance used for generating <code>Set-Cookie</code> headers
+     */
+    public CookieCompliance getSetCookieCompliance()
```

Comment:
It's unfortunate that its `Set-Cookie`, as this line is very confusing now.
`getSet<name>` isn't very clear.

Reply:
Perhaps `getResponseSetCookieCompliance` is more clear.

Reply:
Nah, `setResponseSetCookieCompliance` - too many `set`, I prefer `generator`.
`Generator`, `generating` is besides what is being written in comment and javadocs naturally, so I think it should be so in the code too.

Reply:
I like `getRequestCookieCompliance` and `getResponseCookieCompliance` OK (no need for the `Set` if scoped by response), but that requires a name change from the existing `getCookieCompliance`.
The benefit of `getSetCookieCompliance` is that we don't need to change the name and have deprecated names etc.

There is documentation in the mod file (in fact your comment saying there is no documentation was against the documentation!)

This will be a rarely changed configuration, it is now well documented, so I think the names that I have are fine.      But if you really insist, I'll go for the names above and deprecate the current  `getCookieCompliance` methods.   I definitely will not use `Parser` and `Generator` as they from implementation name space and meaningless to users.

Reply:
Also, "Parser" and "Generator" has different meanings from a Server or Client point of view (eg: a possible future where Jetty HttpClient uses the `CookieCompliance` object)

Reply:
@gregw @joakime "request" and "response" assume a server-side point of view.
For a client receiving a response, you want the _parser_ cookie compliance, and the generator cookie compliance will never be used.
So I maintain that "parser" and "generator" are the right names. 
However, being `HttpConfiguration` a server-side object unlikely to be used on the client side I can live with "request" and "response" even if they are less correct.

Just not "setset" and "getset" that are too confusing.